### PR TITLE
Use standard style in `handle_dotty_assignment()`

### PR DESCRIPTION
**Pull request by lionel-**
What do you think about this more standard style @kevinushey?

- Only create errors when something goes wrong (@DavisVaughan mentions this avoids performance issues with backtrace creation in some cases).

- Use standard constructors like `Ok()` and `Some()`.

-  If we're not interested in a value except for exiting, use explicit exits instead of `?`. I feel like the latter is the equivalent of doing `predicate() || return()` instead of the clearer `if (predicate()) return()`

### CODE REVIEW

File: crates/ark/src/lsp/diagnostics.rs [hunk 1]
```diff
@@ -171,42 +171,52 @@ fn is_within_magrittr_pipe(node: Node, context: &mut DiagnosticContext) -> bool
...
+    if !lhs.is_subset() {
+        return None;
+    };
+
+    let function = lhs.child_by_field_name("function")?;
+    if !function.is_identifier_or_string() {
+        return None;
+    };
+
+    let value = context.contents.node_slice(&function).ok()?.to_string();
```
Comment (DavisVaughan) [inline crates/ark/src/lsp/diagnostics.rs:185]: I have a slight preference for actually logging the potential error here and manually doing `return None`, like in an `Err` match arm

Comment (DavisVaughan) [reply crates/ark/src/lsp/diagnostics.rs:185]: Ah I see how often you use it below though, hmm.

**Review by DavisVaughan — APPROVED** [review]
I generally find this much easier to read

**Review by kevinushey — APPROVED** [review]
LGTM! I wasn't aware (or had forgotten?) about the potential performance costs of backtrace creation as welll.

**Pull request merged**